### PR TITLE
Forge tests for Softmax operator

### PR DESCRIPTION
**Pull request by vobojevicTT**
Add tests for Forge Softmax operator.

### CODE REVIEW

File: forge/test/operators/pytorch/nn/test_softmax.py [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+class ModelFromAnotherOp(torch.nn.Module):
+
+    model_name = "model_op_src_from_another_op"
+
+    def __init__(self, shape, dim):
+        super(ModelFromAnotherOp, self).__init__()
+        self.testname = "Softmax_operator_test_op_src_from_another_op"
+        self.shape = shape
+        self.dim = dim
+        self.softmax = torch.nn.Softmax(dim=self.dim)
```
Comment (vbrkicTT) [inline forge/test/operators/pytorch/nn/test_softmax.py:80]: @nvukobratTT 
Here we use class variant of softmax https://pytorch.org/docs/stable/generated/torch.nn.Softmax.html. There is also functional variant https://pytorch.org/docs/stable/generated/torch.nn.functional.softmax.html.
Class variant is basically wrapper around functional so it probably doesn't make difference which we select.
Maybe package `nn` is too wide here. I suggest we should something more specific like use `activations`.

Comment (nvukobratTT) [reply forge/test/operators/pytorch/nn/test_softmax.py:80]: I don't have hard preferences on this one. Using either class definition or functional one would output the same results compiler wise :)) 

From a model perspective, it's common to see both of these, so let's choose one of the approaches for our testing, and if needed we can extend it for other definitions as well (but this is unlikely)

**Review by dgolubovicTT — APPROVED** [review]
Legit!

File: forge/test/operators/pytorch/nn/test_softmax.py [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+    model: Type[torch.nn.Module],
+    dim: int,
+    input_shape: TensorShape,
+    number_of_operands: int,
+    input_params: List[Dict] = [],
+    input_source_flag: InputSourceFlags = None,
+    dev_data_format: forge.DataFormat = None,
+    math_fidelity: forge.MathFidelity = None,
+):
+    """Common verification function for all tests"""
```
[the remark this answers is not in the document]
Comment (dgolubovicTT) [reply forge/test/operators/pytorch/nn/test_softmax.py:150]: @vobojevicTT any thoughts?

File: forge/test/operators/pytorch/nn/test_softmax.py (hunk 1 above)
Comment (vobojevicTT) [reply forge/test/operators/pytorch/nn/test_softmax.py:80]: I agree that nn package is too wide to organize our tests in that way. We should consider to reorganize tests in another PR.

File: forge/test/operators/pytorch/nn/test_softmax.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vobojevicTT) [reply forge/test/operators/pytorch/nn/test_softmax.py]: Not sure why, but I missed this comment before.
This 'code' comment was misleading. Common functionality is shared via `VerifyUtils.verify`. Here we have an opportunity to adjust tensors and verification parameters per operator.
